optimade/models/structures.py

```diff
@@ -646,15 +646,24 @@ def required_if_dimension_types_has_one(cls, v, values):
                 f"lattice_vectors is REQUIRED, since dimension_types is not [0, 0, 0] but is {values.get('dimension_types', 'Not specified')}"
             )
 
-        for dim_type, vector in zip(values.get("dimension_types", []), v):
+        for dim_type, vector in zip(values.get("dimension_types", (None,) * 3), v):
             if None in vector and dim_type == 1:
```

Comment:
use the enum value: `Periodicity.APERIODIC` to be resilient to convention changes in the future.

Reply:
Yup, and as you write in #170 we do indeed have this validator. I actually started writing another validator (using `Periodicity`) but found that this was already there and then lazily didn't bother updating it - will update now though, since you've so sharply spotted the use case for `Periodicity` :smile: